### PR TITLE
Add output defs for mode kernel

### PR DESCRIPTION
**Pull request by ccsuzzh**
### PR types

Others
### PR changes

Others
### Describe

#51292 case43 mode

### CODE REVIEW

**Comment by luotao1** [comment]
@From00 可以进行 review

File: paddle/phi/kernels/cpu/mode_kernel.cc [hunk 1]
```diff
@@ -132,4 +132,6 @@ void ModeKernel(const Context& dev_ctx,
 }  // namespace phi
 
 PD_REGISTER_KERNEL(
-    mode, CPU, ALL_LAYOUT, phi::ModeKernel, float, double, int32_t, int64_t) {}
+    mode, CPU, ALL_LAYOUT, phi::ModeKernel, float, double, int32_t, int64_t) {
+  kernel->OutputAt(1).SetDataType(paddle::experimental::DataType::INT64);
```
Comment (From00) [inline paddle/phi/kernels/cpu/mode_kernel.cc]:
```suggestion
  kernel->OutputAt(1).SetDataType(phi::DataType::INT64);
```

Comment (ccsuzzh) [reply paddle/phi/kernels/cpu/mode_kernel.cc]: done

**Review by From00 — APPROVED** [review]
LGTM

**Pull request merged**